File: app/src/main/java/fi/masaz/celatum/room/Item.java
package fi.masaz.celatum.room;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity
public class Item {
    @PrimaryKey(autoGenerate = true)
    public int id;

    @ColumnInfo(name = "title")
    public String title;

    @ColumnInfo(name = "description")
    public String description;

    @ColumnInfo(name = "icon")
    public String icon;

    @ColumnInfo(name = "created_ts")
    public Long createdTs;

    @ColumnInfo(name = "edited_ts")
    public Long editedTs;
}

File: app/src/main/java/fi/masaz/celatum/room/Item.kt
package fi.masaz.celatum.room

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
class Item {
    @PrimaryKey(autoGenerate = true)
    var id: Int = 0

    @ColumnInfo(name = "title")
    var title: String? = null

    @ColumnInfo(name = "description")
    var description: String? = null

    @ColumnInfo(name = "icon")
    var icon: String? = null

    @ColumnInfo(name = "created_ts")
    var createdTs: Long? = null

    @ColumnInfo(name = "edited_ts")
    var editedTs: Long? = null
}
